pkg/master/master.go

```diff
@@ -544,7 +554,34 @@ func (m *Master) init(c *Config) {
 	}
 
 	// TODO: Attempt clean shutdown?
-	m.masterServices.Start()
+	if m.enableCoreControllers {
+		m.NewBootstrapController().Start()
+	}
+}
+
+// NewBootstrapController returns a controller for watching the core capabilities of the master.
+func (m *Master) NewBootstrapController() *Controller {
```

Comment:
That's what we would use to get the default implementation to tweak.

I.e. OpenShift says:

```
m := master.New(config)
c := m.NewBootstrapController()
c.<some stuff> = <some other stuff>
c.Run()

----- Original Message -----
> > @@ -544,7 +554,34 @@ func (m *Master) init(c *Config) {
> >  	}
> >  
> >  	// TODO: Attempt clean shutdown?
> > -	m.masterServices.Start()
> > +	if m.enableCoreControllers {
> > +		m.NewBootstrapController().Start()
> > +	}
> > +}
> > +
> > +// NewBootstrapController returns a controller for watching the core
> > capabilities of the master.
> > +func (m *Master) NewBootstrapController() *Controller {
> 
> unexport this?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7704/files#r29889869